### PR TITLE
[Feature] Add TIMM and HuggingFace wrappers to build classifiers from them directly.

### PR DESCRIPTION
**Pull request by mzr1996**
## Motivation

Pytorch-image-models and HuggingFace are famous model hubs, this PR can integrate their models in mmcls and use mmcls to train and inference models come from timm and huggingface directly.

### To train a model from TIMM
For example, to train a ResNet-50 model from timm with our vit-mae schedule.
Use config file as below
```python
# configs/resnet/resnet50_timm.py
_base_ = ['../vision_transformer/vit-base-p16_pt-32xb128-mae_in1k-224.py']

model = dict(
    type='TimmClassifier',
    model_name='resnet50',
    loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
    train_cfg=dict(augments=[
        dict(type='Mixup', alpha=0.8),
        dict(type='CutMix', alpha=1.0)
    ]),
    _delete_=True)
```
### To fine-tune a model from hugging-face
For example, to fine-tune a ResNet-50 model from hugging face on CIFAR dataset.
Use config file as below
```python
# configs/resnet/resnet50_huggingface_cifar.py
_base_ = [
    '../_base_/datasets/cifar10_bs16.py',
    '../_base_/schedules/cifar10_bs128.py',
    '../_base_/default_runtime.py',
]

model = dict(
    type='HuggingFaceClassifier',
    model_name='microsoft/resnet-50',
    pretrained=True,
    loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
    num_labels=10,
    ignore_mismatched_sizes=True,
)
```

### To inference a TIMM model
```python
>>> from mmengine import Config
>>> from mmcls.apis import init_model, inference_model
>>> from mmcls.utils import register_all_modules
>>> from mmcls.datasets.categories import IMAGENET_CATEGORIES
>>> register_all_modules()
>>> # Use data pipeline from the resnet config
>>> cfg = Config.fromfile("./configs/resnet/resnet50_8xb32_in1k.py")
>>> cfg.model = dict(type='TimmClassifier', model_name='resnet50', pretrained=True)
>>> model = init_model(cfg)
>>> pred_label = inference_model(model, 'demo/demo.JPEG')['pred_label']
>>> print(IMAGENET_CATEGORIES[pred_label])
'sea snake'
```
### To inference a HuggingFace model
```python
>>> from mmengine import Config
>>> from mmcls.apis import init_model, inference_model
>>> from mmcls.utils import register_all_modules
>>> from mmcls.datasets.categories import IMAGENET_CATEGORIES
>>> register_all_modules()
>>> # Use data pipeline from the resnet config
>>> cfg = Config.fromfile("./configs/resnet/resnet50_8xb32_in1k.py")
>>> cfg.model = dict(type='HuggingFaceClassifier', model_name='microsoft/resnet-50', pretrained=True)
>>> model = init_model(cfg)
>>> pred_label = inference_model(model, 'demo/demo.JPEG')['pred_label']
>>> print(IMAGENET_CATEGORIES[pred_label])
'sea snake'
```

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcls/models/classifiers/hugging_face.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        from transformers import AutoConfig, AutoModelForImageClassification
+        if pretrained:
+            self.model = AutoModelForImageClassification.from_pretrained(
+                model_name, *model_args, **kwargs)
+        else:
+            config = AutoConfig.from_pretrained(model_name, *model_args,
+                                                **kwargs)
+            self.model = AutoModelForImageClassification.from_config(config)
+        self.loss_module = MODELS.build(loss)
+
```
Comment (okotaku) [inline mmcls/models/classifiers/hugging_face.py:47]:
```
if with_cp:
    self.model.gradient_checkpointing_enable()
```
It can support gradient checkpointing in transformers.

File: mmcls/models/classifiers/timm.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        if train_cfg is not None and 'augments' in train_cfg:
+            # Set batch augmentations by `train_cfg`
+            data_preprocessor['batch_augments'] = train_cfg
+
+        super().__init__(
+            init_cfg=init_cfg, data_preprocessor=data_preprocessor)
+        from timm.models import create_model
+        self.model = create_model(*args, **kwargs)
+        self.loss_module = MODELS.build(loss)
+
```
Comment (okotaku) [inline mmcls/models/classifiers/timm.py:38]:
```
If with_cp:
    self.model.set_grad_checkpointing()
```

timm also supports gradient checkpointing in latest versions. You may want to consider this.

Comment (mzr1996) [reply mmcls/models/classifiers/timm.py:83]: Updated
